mm2src/coins/utxo/qrc20.rs

```diff
@@ -0,0 +1,690 @@
+        utxo_common::get_verbose_transaction_from_cache_or_rpc(self, ctx, txid).await
+    }
+
+    fn is_unspent_mature(&self, output: &RpcTransaction) -> bool {
+        qtum::is_qtum_unspent_mature(self.arc().mature_confirmations, output)
+    }
+}
+
+async fn qrc20_withdraw(coin: Qrc20Coin, ctx: MmArc, req: WithdrawRequest) -> Result<TransactionDetails, String> {
+    let to_addr = match &coin.arc().address_format {
+        UtxoAddressFormat::Standard => try_s!(Address::from_str(&req.to)),
+        UtxoAddressFormat::CashAddress {..} => try_s!(Address::from_cashaddress(
+            &req.to, coin.arc().checksum_type.clone(), coin.arc().pub_addr_prefix, coin.arc().p2sh_addr_prefix))
+    };
+
+    let is_p2pkh = to_addr.prefix == coin.arc().pub_addr_prefix && to_addr.t_addr_prefix == coin.arc().pub_t_addr_prefix;
+    let is_p2sh = to_addr.prefix == coin.arc().p2sh_addr_prefix && to_addr.t_addr_prefix == coin.arc().p2sh_t_addr_prefix && coin.arc().segwit;
+    if !is_p2pkh && !is_p2sh {
+        return ERR!("Address {} has invalid format", to_addr);
+    }
+
+    let _utxo_lock = UTXO_LOCK.lock().await;
+
+    // the qrc20_amount is used only within smart contract calls
+    let qrc20_amount = if req.max {
+        let balance = try_s!(coin.my_balance().compat().await);
+        try_s!(wei_from_big_decimal(&balance, coin.arc().decimals))
+    } else {
+        try_s!(wei_from_big_decimal(&req.amount, coin.arc().decimals))
+    };
+
+    let (gas_limit, gas_price) = match req.fee {
+        Some(WithdrawFee::Qrc20Gas { gas_limit, gas_price }) => (gas_limit, gas_price),
+        Some(_) => return ERR!("Unsupported input fee type"),
+        None => (QRC20_GAS_LIMIT_DEFAULT, QRC20_GAS_PRICE_DEFAULT),
+    };
+
+    let script_pubkey = try_s!(generate_token_transfer_script_pubkey(
+        to_addr.clone(), qrc20_amount, gas_limit, gas_price, &coin.contract_address)).to_bytes();
+
+    // qtum_amount is always 0 for the QRC20, because we should pay only a fee in Qtum to send the QRC20 transaction
+    let qtum_amount = 0u64;
+    let outputs = vec![TransactionOutput {
+        value: qtum_amount,
+        script_pubkey,
+    }];
+
+    let unspents = try_s!(coin.ordered_mature_unspents(&ctx, &coin.arc().my_address).compat().await.map_err(|e| ERRL!("{}", e)));
+
+    // None seems that the generate_transaction() should request estimated fee for Kbyte
+    let actual_tx_fee = None;
+    let gas_fee = Some(gas_limit * gas_price);
+    let fee_policy = FeePolicy::SendExact;
+
+    let (unsigned, data) = try_s!(coin.generate_transaction(unspents, outputs, fee_policy, actual_tx_fee, gas_fee).await);
+    let prev_script = Builder::build_p2pkh(&coin.arc().my_address.hash);
+    let signed = try_s!(sign_tx(unsigned, &coin.arc().key_pair, prev_script, coin.arc().signature_version, coin.arc().fork_id));
+    let fee_details = UtxoFeeDetails {
```

Comment:
UtxoFeeDetails is expected that fee is paid in coin that is withdrawn, but it is not the case for QRC20. Seems like it should be mix of UtxoFeeDetails and EthFeeDetails, e.g.:
```
coin: QTUM, // QRC20 transfer fee is paid in QTUM
miner_fee: standard UTXO miner fee based on transaction size,
gas_price: 40, // satoshis
gas_limit: 250000,
total_gas_fee: 0.1
```